wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/wicket-ajaxdownload.js

```diff
@@ -28,7 +28,7 @@
 	Wicket.AjaxDownload = {
 		initiate : function(settings) {
 			document.cookie = settings.name +
-				'=;path=/;Max-Age=0;expires=Thu, 01 Jan 1970 00:00:01 GMT';
+				'=;path=/;Max-Age=0;expires=Thu, 01 Jan 1970 00:00:01 GMT; SameSite=None; Secure';
```

Comment:
I'm not sure regarding this change

One of use-cases of our application is "Run inside `iframe`"
This mode requires modification of `webapps/_ctx_/META-INF/context.xml`
And setting

`<CookieProcessor sameSiteCookies="None" />` (Lax by default)

So _maybe_ Servlet container can handle this?

Reply:
> I'm not sure regarding this change
> 
> One of use-cases of our application is "Run inside `iframe`" This mode requires modification of `webapps/_ctx_/META-INF/context.xml` And setting
> 
> `<CookieProcessor sameSiteCookies="None" />` (Lax by default)
> 
> So _maybe_ Servlet container can handle this?

I don't think so becase this is used by server side cookie processor like in code bellow 

![image](https://user-images.githubusercontent.com/462655/229450273-ea8247c1-be69-4c68-ac10-325f6b085d6e.png)

to add the SameSite thing to server side generated cookies. Maybe the we should make this configurable by passing some paramter to the JavaScrip layer?

Reply:
Code above is from

![image](https://user-images.githubusercontent.com/462655/229451279-2930818e-9505-497d-a3d4-5404cb30f6f6.png)

Reply:
I just created https://issues.apache.org/jira/browse/WICKET-7038
Servlet API 6.x has the setter.
But this is only for Wicket 10.x

Reply:
A cookie is just some special header... Maybe this is teh approach we should follow here so that we can have the same at client side and server side. Because if we use setCookie then, at least in case fo tomcat, application sever will use its own machiney to write SameSite attibuture (e.g. see Rfc6265CookieProcessor)